src/components/TimelineComponent.tsx

```diff
@@ -25,20 +59,187 @@ export default class TimelineComponent extends
       data: props.data,
       width: window.innerWidth,
       height: window.innerHeight,
-      marginTop: 40,
-      marginBottom: 100,
+      marginTop: 0,
+      marginBottom: 170,
       marginLeft: 40,
       marginRight: 20,
+      toggleTimeline: 0,
+      togglePrompt: 'Switch to Interval Timeline',
+      yColumn: '',
+      xColumn: '',
+      xColumn2: '',
+      loading: true,
     };
 
     this.drawTimeline = this.drawTimeline.bind(this);
+    this.toggleTimeline = this.toggleTimeline.bind(this);
+    this.initTimeline = this.initTimeline.bind(this);
+    this.ttOverHelper = this.ttOverHelper.bind(this);
+    this.ttOver = this.ttOver.bind(this);
+    this.ttUpdatePos = this.ttUpdatePos.bind(this);
+    this.ttMove = this.ttMove.bind(this);
+    this.ttMove = this.ttMove.bind(this);
+    this.updateChart = this.updateChart.bind(this);
+    this.updateBars = this.updateBars.bind(this);
+    this.moveChart = this.moveChart.bind(this);
+    this.drawEventMagnitude = this.drawEventMagnitude.bind(this);
+    this.drawIntervalMagnitude = this.drawIntervalMagnitude.bind(this);
+    this.dragStarted = this.dragStarted.bind(this);
+    this.dragged = this.dragged.bind(this);
+    this.dragEnded = this.dragEnded.bind(this);
+    this.resetTimeline = this.resetTimeline.bind(this);
+    this.sortData = this.sortData.bind(this);
+    // this.changeColumn = this.changeyColumn.bind(this);
   }
 
+
   /**
    * Purpose: waits until the component has properly mounted before drawing the
    * timeline
    */
   componentDidMount(): void {
+    // console.log(this.state.data);
+
```

Comment:
for longer functions such as componentDidMount() some more inline commenting would be usefule